api/src/main/java/io/minio/S3Base.java

```diff
@@ -145,7 +146,8 @@ protected S3Base(
       boolean useVirtualStyle,
       String region,
       Provider provider,
-      OkHttpClient httpClient) {
+      OkHttpClient httpClient,
+      boolean closeHttpClient) {
```

Comment:
This breaks backward compatibility.

You would need to create another constructor and have this `closeHttpClient` argument and call the newly added constructor in this constructor.

Finally deprecate this constructor.

Reply:
good catch, fixed!